shanghai/core_plugins/channel.py

```diff
@@ -19,10 +19,11 @@
 import re
 import string
 
-from shanghai.event import global_event, GlobalEventName
-from shanghai.network import NetworkContext
+from shanghai.event import global_event, GlobalEventName, EventDispatcher, Priority
+from shanghai.network import ShadowAttributesMixin, NetworkContext
 from shanghai.irc.server_reply import ServerReply
 from .message import Message
```

Comment:
This import should probably be separated from the others. Otherwise, it should be made absolute as well.